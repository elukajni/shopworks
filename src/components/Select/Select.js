import React from 'react'
import "./Select.scss";

const Select = ({ fieldData, value, handleChange, parent }) => {
    return (
        <select name={fieldData.name}
            onChange={event => handleChange(fieldData.name, event.target.value, parent)}
            value={value}
            className="selectInput"
        >
            {fieldData.options.map((option, index) =>
                <option value={option} key={index}>{option}</option>
            )}
        </select>)
}

export default Select