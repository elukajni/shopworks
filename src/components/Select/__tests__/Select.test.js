import React from 'react';
import renderer from 'react-test-renderer';

import Select from "../Select"
function createTestProps() {
    return {
        fieldData: { "name": "country", "label": "Country", "type": "dropdown", "options": ["Canada", "USA"] },
        value: "",
        parent: "",
        handleChange: () => { }
    }
}
let props
beforeEach(() => {
    props = createTestProps();
});

it('renders correctly', () => {
    const tree = renderer
        .create(<Select {...props} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});