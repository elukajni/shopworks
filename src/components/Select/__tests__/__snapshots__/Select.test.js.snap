// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<select
  className="selectInput"
  name="country"
  onChange={[Function]}
  value=""
>
  <option
    value="Canada"
  >
    Canada
  </option>
  <option
    value="USA"
  >
    USA
  </option>
</select>
`;
